src/lib.rs

```diff
@@ -191,25 +198,41 @@ impl Element {
                         name: name.local_name,
                         attributes: attr_map,
                         children: Vec::new(),
-                        text: None,
                     };
-                    return build(&mut reader, root);
+                    root_nodes.push(XMLNode::Element(build(&mut reader, root)?));
                 }
-                Ok(XmlEvent::Comment(..))
-                | Ok(XmlEvent::Whitespace(..))
-                | Ok(XmlEvent::StartDocument { .. }) => continue,
-                Ok(XmlEvent::EndDocument)
-                | Ok(XmlEvent::EndElement { .. })
-                | Ok(XmlEvent::Characters(..))
-                | Ok(XmlEvent::CData(..))
-                | Ok(XmlEvent::ProcessingInstruction { .. }) => {
-                    return Err(ParseError::CannotParse)
+                Ok(XmlEvent::Comment(comment_string)) => {
+                    println!("comment");
```

Comment:
Debugging code here to be removed

Reply:
Jumped on it right as I found that error!